src/components/templates/product-availability-modal/sales-channel-availability-table.tsx

```diff
@@ -0,0 +1,228 @@
+    pageCount,
+    nextPage,
+    previousPage,
+    // Get the state from the instance
+    state: { pageIndex, ...state },
+  } = tableState
+
+  React.useEffect(() => {
+    if (setSelectedRowIds) {
+      setSelectedRowIds(Object.keys(state.selectedRowIds))
+    }
+  }, [Object.keys(state.selectedRowIds).length])
+
+  const handleNext = () => {
+    if (canNextPage) {
+      setOffset(offset + limit)
+      nextPage()
+    }
+  }
+
+  const handlePrev = () => {
+    if (canPreviousPage) {
+      setOffset(Math.max(offset - limit, 0))
+      previousPage()
+    }
+  }
+
+  return (
+    <div className="min-h-[350px] flex flex-col justify-between">
+      <Table
+        {...getTableProps()}
+        enableSearch
+        handleSearch={setQuery}
+        tableActions={tableAction}
+      >
+        <Table.Head>
+          {headerGroups?.map((headerGroup) => (
+            <Table.HeadRow {...headerGroup.getHeaderGroupProps()}>
+              {headerGroup.headers.map((col) => (
+                <Table.HeadCell {...col.getHeaderProps()}>
+                  {col.render("Header")}
+                </Table.HeadCell>
+              ))}
+            </Table.HeadRow>
+          ))}
+        </Table.Head>
+        <Table.Body {...getTableBodyProps()}>
+          {rows.map((row) => {
+            prepareRow(row)
+            return (
+              <Table.Row color={"inherit"} {...row.getRowProps()}>
+                {row.cells.map((cell, index) => {
+                  return (
+                    <Table.Cell {...cell.getCellProps()}>
+                      {cell.render("Cell", { index })}
+                    </Table.Cell>
+                  )
+                })}
+              </Table.Row>
+            )
+          })}
+        </Table.Body>
+      </Table>
+      <TablePagination
+        count={salesChannels.length!}
+        limit={limit}
+        offset={offset}
+        pageSize={offset + rows.length}
+        title="Sales Channels"
+        currentPage={pageIndex + 1}
+        pageCount={pageCount}
+        nextPage={handleNext}
+        prevPage={handlePrev}
+        hasNext={canNextPage}
+        hasPrev={canPreviousPage}
+      />
+    </div>
+  )
+}
+
+export const SalesChannelTableActions: React.FC<SalesChannelTableActionsProps> = ({
+  numberOfSelectedRows,
+  availableChannelIds,
+  onAddSalesChannelsToAvailableChannels,
+  onDeselect,
+  onRemove,
+}) => {
+  const addChannelModalScreen = useAddChannelsModalScreen(
+    availableChannelIds,
+    onAddSalesChannelsToAvailableChannels
+  )
+
+  const showAddChannels = !!numberOfSelectedRows
+
+  const classes = {
+    "translate-y-[-42px]": !showAddChannels,
+    "translate-y-[0px]": showAddChannels,
+  }
+
+  const { push } = React.useContext(LayeredModalContext)
+
+  return (
+    <div className="flex space-x-xsmall h-[34px] overflow-hidden">
+      <div className={clsx("transition-all duration-200", classes)}>
+        <div className="divide-x flex items-center mb-2">
```

Comment:
todo: should we add a height of 84 on this component or heights of 42 to both children?

Reply:
I have limited the heights a bit more due to the border highlighting on the `add-channels` button since the border would show over the 'remove' and 'deselect' buttons of the view after the transition. But I have a added heights to each of the children as well